six/modules/c++/six.sicd/source/Utilities.cpp

```diff
@@ -489,6 +489,24 @@ void Utilities::getWidebandData(
             offset, extent, buffer);
 }
 
+Vector3 Utilities::getGroundPlaneNormal(const ComplexData& data)
+{
+    Vector3 groundPlaneNormal;
+
+    if (data.radarCollection->area.get() &&
+        data.radarCollection->area->plane.get())
+    {
+        AreaPlane areaPlane = *data.radarCollection->area->plane;
+        groundPlaneNormal = cross(areaPlane.xDirection->unitVector,
```

Comment:
All the changes are made.

From: Adam Sylvester [mailto:notifications@github.com]
Sent: Monday, May 09, 2016 7:53 AM
To: ngageoint/six-library <six-library@noreply.github.com>
Cc: Jonathan Means <Jonathan.Means@mdaus.com>; Author <author@noreply.github.com>
Subject: Re: [ngageoint/six-library] Add getGroundPlaneNormal() to SICD Utilities (#70)


In six/modules/c++/six.sicd/source/Utilities.cpp<http://cp.mcafee.com/d/FZsSd1MA76Qm74NNEVo76zAQsCXCSm6hMUQsLLCTNPb3a9EVd79J6XRTD3hPsQsL6QkmkkTPhOUM-r4qMSpfosH-ME2BkdrFOVJl3mWsKrtjQttAs_R-jvup7ccZuVtdcQsLYMUOehu7th5dqWqJSmel3PWApmU6CQjqpK_8K6zBV55BeXNKVIDeqR4IMzapdIxO-g8Wh-IEjOAwYKv8x7oBImrffMts349kITiAGM5oK9Ife8CzBYT2Ivmfb0EzapdIxOVJZBAQsCMnWhEwdbs2eAvHa4YF8fbDO8hS9oQKCy0gggd417VxadmU81o_ou3aNlQYQk6yAQgjGq81wd4wUQg1eszVjBZ3h0qa0c7nd59IsYrev8k>:

> @@ -489,6 +489,24 @@ void Utilities::getWidebandData(

>              offset, extent, buffer);

>  }

>

> +Vector3 Utilities::getGroundPlaneNormal(const ComplexData& data)

> +{

> +    Vector3 groundPlaneNormal;

> +

> +    if (data.radarCollection->area.get() &&

> +        data.radarCollection->area->plane.get())

> +    {

> +        AreaPlane areaPlane = *data.radarCollection->area->plane;

> +        groundPlaneNormal = cross(areaPlane.xDirection->unitVector,

Hmm, I'm actually confused how the compiler knows how to resolve that... I don't see a using namespace math::linear anywhere. Anyway yeah, still stick the math::linear:: in front of it to be explicit.

—
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub<http://cp.mcafee.com/d/5fHCNAi6hAqdEIe9zzhOMed79EVdTdIIczxNEVvvdLzCm6kjhOqejqdTHLe6zCVEVudEEIEFLCzBNxYS8RxIOuMVnZxg5aEqTjBPqG6JQVsSWDEWX8V_HYC-YOeopWZOWqpEVvVxNAsyYeWyaqRQRrIIsG7DR8OJMddFCQPt-hsd7bOabatTztPpesRG9px6kOrp3BYwhQzZpgDB91Vs-h2eNboISuvwW-eOf-CcmZwwUwZKK3CrE8cfmd7bOa23H3y26IIK9Knh78f0p5NdxVN4QsLCUlzWNVo54pj9JAendLIICzAS2_id41FrwhQzZpgDB91Vs-h2eNb6BQQg2221Ew8_c9hGT10b7X3MpmaKDCywQkCy2tjh0c1EA76y09PAvasLEq83hg1wWVEFdzDzpHLQE-aCpEGK>


The information contained in this communication is confidential, is intended only for the use of the recipient(s) named above, and may be legally privileged. If the reader of this message is not the intended recipient, you are hereby notified that any dissemination, distribution, or copying of this communication is strictly prohibited. If you have received this communication in error, please re-send this communication to the sender and delete the original message or any copy of it from your computer system.